kratos/includes/properties.h

```diff
@@ -278,31 +293,383 @@ class Properties : public IndexedObject
+     * @return The pointer to the subproperty of interest
+     */
+    KRATOS_DEPRECATED_MESSAGE("This is legacy version. Use pGetSubPropertiesById") Properties::Pointer pGetSubProperty(const IndexType SubPropertyIndex) const
+    {
+        return pGetSubPropertiesById(SubPropertyIndex);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    KRATOS_DEPRECATED_MESSAGE("This is legacy version. Use GetSubPropertiesById") Properties& GetSubProperty(const IndexType SubPropertyIndex)
+    {
+        return GetSubPropertiesById(SubPropertyIndex);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id (constant version)
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    KRATOS_DEPRECATED_MESSAGE("This is legacy version. Use GetSubPropertiesById") Properties& GetSubProperty(const IndexType SubPropertyIndex) const
+    {
+        return GetSubPropertiesById(SubPropertyIndex);
+    }
+
+    /**
+     * @brief This method checks if the subproperty exists from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return True if there is such subproperty, false otherwise
+     */
+    bool HasSubPropertiesById(const IndexType SubPropertyIndex)
+    {
+        // Checking if the id is the current properties
+        if (this->Id() == SubPropertyIndex) {
+            KRATOS_WARNING("Properties") << "There is no subproperty with the Id: " << SubPropertyIndex << ". The current property has that Id" << std::endl;
+            return true;
+        }
+
+        // Looking into the database
+        auto property_iterator = mSubPropertiesList.find(SubPropertyIndex);
+        if (property_iterator != mSubPropertiesList.end()) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @brief This method checks if the subproperty exists from the index corresponding to the property id  (constant version)
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return True if there is such subproperty, false otherwise
+     */
+    bool HasSubPropertiesById(const IndexType SubPropertyIndex) const
+    {
+        // Checking if the id is the current properties
+        if (this->Id() == SubPropertyIndex) {
+            KRATOS_WARNING("Properties") << "There is no subproperty with the Id: " << SubPropertyIndex << ". The current property has that Id" << std::endl;
+            return true;
+        }
+
+        // Looking into the database
+        auto property_iterator = mSubPropertiesList.find(SubPropertyIndex);
```

Comment:
```suggestion
        return mSubPropertiesList.find(SubPropertyIndex) != mSubPropertiesList.end();
```
reduces 5 lines to 1 line :D

Reply:
OK